junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/execution/ParameterResolutionUtils.java

```diff
@@ -99,7 +99,6 @@ public static Object[] resolveParameters(Executable executable, Optional<Object>
 		// Ensure that the outer instance is resolved as the first parameter if
 		// the executable is a constructor for an inner class.
 		if (outerInstance.isPresent()) {
-			Preconditions.condition(parameters[0].isImplicit(), "First parameter must be implicit");
```

Comment:
Instead of deleting this check, wouldn't it be better to also apply the same "isInnerClass(...)" test?

Reply:
That's basically what `outerInstance.isPresent()` does. I only recently added this check when working on parameterized classes.